### ISSUE TITLE
Vector should be representational

### ISSUE THREAD

**Issue by idontgetoutmuch** [haskell/vector#223]
At the moment we get this

```
*Numeric.Sundials.CVode.ODE> (coerce ((V.fromList [1,2]) :: Vector CInt)) :: Vector Int
[8589934593,4426664329]
```

We should really get a type error.

**Comment by idontgetoutmuch** [comment]
And by making Vector representational we get

```
*Data.Vector> (coerce ((Data.Vector.fromList [1,2]) :: Vector CInt)) :: Vector Int

<interactive>:9:2: error:
    • Couldn't match representation of type ‘GHC.Int.Int32’
                               with that of ‘Int’
        arising from a use of ‘coerce’
    • In the expression:
          (coerce ((Data.Vector.fromList [1, 2]) :: Vector CInt)) ::
            Vector Int
      In an equation for ‘it’:
          it
            = (coerce ((Data.Vector.fromList [1, 2]) :: Vector CInt)) ::
                Vector Int
```

**Comment by RyanGlScott** [comment]
Fixed in #224.
